lib/parser.js

```diff
@@ -1,41 +1,184 @@
 'use strict';
 
-const HANDSHAKE_LENGTH = 36;
-const PARCEL_LENGTH = 16;
-const CHUNK_LENGTH = 12;
+const constants = {
+  PROTOCOL_VERSION: 1,
+  HANDSHAKE_LENGTH: 36,
+  PARCEL_LENGTH: 16,
+  CHUNK_LENGTH: 12,
+  STRUCT_PARCEL: 0,
+  STRUCT_CHUNK: 1,
+  PARCEL_PING: 0,
+  PARCEL_PONG: 1,
+  PARCEL_CALL: 2,
+  PARCEL_CALLBACK: 3,
+  PARCEL_EVENT: 4,
+  PARCEL_STREAM: 5
+};
 
-const parcel = (buffer) => {
+const readParcel = (
   // Read parcel from buffer
-  // Return parcel object
-  const rest = Buffer.from(buffer, 1, PARCEL_LENGTH);
-  return rest;
+  buffer // buffer
+  // Return: object
+) => {
+  const rest = buffer.slice(0, constants.PARCEL_LENGTH);
+  const parcel = {
+    structType: constants.STRUCT_PARCEL,
+    parcelId: rest.readInt32LE(1),
+    parcelType: rest.readInt8(5),
+    compression: rest.readInt8(6),
+    encoding: rest.readInt8(7),
+    length: parseInt(rest.slice(8).toString()),
+  };
+  return parcel;
+};
+
+const parcel = ({
+  // Encode parcel structure to Buffer
+  parcelId, // number, id of parcel: 4b
```

Comment:
Looking at this comment, I can assume that `parcelId`s' value will always be `4b`,, not that its size is 4b. Change it to more appropriate or remove at all